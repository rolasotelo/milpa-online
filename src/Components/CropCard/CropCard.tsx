import React from "react";

interface Props {
  title: string;
}

const CropCard = (props: Props) => {
  return <div className="bg-mexicanBone m-2 h-20">{props.title}</div>;
};

export default CropCard;
