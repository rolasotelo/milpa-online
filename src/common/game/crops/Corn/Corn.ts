import { Crop } from "../../../types";

export const CORN_ID = "corn";

export const Corn: Crop = {
  id: CORN_ID,
  type: "crop",
  name: "Corn",
  icon: "üåΩ",
  description: "El cultivo m√°s importante de mesoam√©rica",
  rules: "WIP",
  resume: "+1 üç´",
  canInteractWith: {
    ownEmptyMilpaSlots: true,
    ownFilledMilpaSlots: ["manure"],
    ownEmptyEdgeSlots: false,
    ownFilledEdgeSlots: false,
    othersEmptyMilpaSlots: false,
    othersFilledMilpaSlots: false,
    othersEmptyEdgeSlots: false,
    othersFilledEdgeSlots: false,
  },
};
