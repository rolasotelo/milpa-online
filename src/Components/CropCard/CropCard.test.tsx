import React from "react";
import renderer from "react-test-renderer";
import { Corn } from "../../common/game/crops/Corn/Corn";

import CropCard from "./CropCard";

describe("<CropCard />", () => {
  const defaultProps = {
    crop: Corn,
  };
  const wrapper = renderer.create(<CropCard {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
