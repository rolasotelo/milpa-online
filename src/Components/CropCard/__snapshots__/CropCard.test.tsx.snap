// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CropCard /> render 1`] = `
<div
  className="bg-mexicanBone  h-28"
>
  Corn
</div>
`;
